client/src/components/Forms/ModalForm.tsx

```diff
@@ -19,10 +19,20 @@ export function ModalForm(props: Props) {
   const antForm = Form.useForm()[0];
   const form = props.form || antForm;
 
+  useEffect(() => {
+    if (props.data && !Object.values(props.data).length) {
+      form.resetFields();
+    }
+  }, [props.data]);
+
   if (props.data == null) {
     return null;
   }
-  const initialValues = props.getInitialValues ? props.getInitialValues?.(props.data) : props.data;
+
+  const formValues = props.getInitialValues ? props.getInitialValues?.(props.data) : props.data;
+
+  form.setFieldsValue(formValues);
```

Comment:
wondering: why to update form manually on each render? what's wrong with initial values? looks kind of buggy

Reply:
Oh, didn't noticed your comment. I think it's related to Antd modal/form weird persistance somehow, it sets proper values from second attempt. We wanted to get rid of this component soon anyway, so I didn't dived too deep into it :)